sdks/python/apache_beam/io/fileio_test.py

```diff
@@ -163,6 +164,43 @@ def test_csv_file_source(self):
 
       assert_that(content_pc, equal_to(rows))
 
+  def test_infer_compressed_file(self):
+    dir = '%s%s' % (self._new_tempdir(), os.sep)
+
+    file_contents = b'compressed_contents!'
+    import gzip
+    with gzip.GzipFile(os.path.join(dir, 'compressed.gz'), 'w') as f:
+      f.write(file_contents)
+
+    with TestPipeline() as p:
+      content_pc = (p
+                    | beam.Create([FileSystems.join(dir, '*')])
+                    | fileio.MatchAll()
+                    | fileio.ReadMatches()
+                    | beam.Map(lambda rf: rf.open().read(len(file_contents))))
+
+      assert_that(content_pc, equal_to([file_contents]))
+
+  def test_read_compressed_file_without_suffix(self):
```

Comment:
Let's also add tests for other compression types we support assuming it is cheap.